zerver/tests/test_message_edit.py

```diff
@@ -1528,6 +1528,36 @@ def check_move_message_to_stream(role: int, error_msg: Optional[str] = None) ->
         )
         check_move_message_to_stream(UserProfile.ROLE_MEMBER)
 
+    def test_move_message_to_stream_with_topic_editing_not_allowed(self) -> None:
+        (user_profile, old_stream, new_stream, msg_id, msg_id_later) = self.prepare_move_topics(
+            "othello", "old_stream_1", "new_stream_1", "test"
+        )
+
+        realm = user_profile.realm
+        realm.allow_community_topic_editing = False
+        realm.save()
+
+        do_set_realm_property(
+            user_profile.realm,
+            "move_messages_between_streams_policy",
+            Realm.POLICY_MEMBERS_ONLY,
+            acting_user=None,
+        )
+
+        result = self.client_patch(
+            "/json/messages/" + str(msg_id),
+            {
+                "message_id": msg_id,
+                "stream_id": new_stream.id,
+                "propagate_mode": "change_all",
+            },
+        )
```

Comment:
Can you add a failed attempt to make this move with `topic` set in the request before this successful one?  That's a good way to validate that the test setup is correct.

Reply:
Added this.